tests/test_builder.py

```diff
@@ -575,6 +575,126 @@ def test_build_runtime_ecr_with_aws_secret(monkeypatch):
         pytest.fail("no create-repo init container")
 
 
+@pytest.mark.parametrize(
+    "dest,registry,default_repository,expected_dest",
+    [
+        (
+            "test-image",
+            None,
+            None,
+            "test-image",
+        ),
+        (
+            "test-image",
+            "test-registry",
+            None,
+            "test-registry/test-image",
+        ),
+        (
+            "test-image",
+            "test-registry/test-repository",
+            None,
+            "test-registry/test-repository/test-image",
+        ),
+        (
+            ".test-image",
+            None,
+            "default.docker.registry/default-repository",
+            "default.docker.registry/default-repository/test-image",
+        ),
+        (
+            ".test-image",
+            None,
+            "default.docker.registry",
+            "default.docker.registry/test-image",
+        ),
+    ],
+)
+def test_get_image_dest(dest, registry, default_repository, expected_dest, monkeypatch):
+    docker_registry_secret = "default-docker-registry-secret"
+    monkeypatch.setattr(config.httpdb.builder, "docker_registry", default_repository)
+    monkeypatch.setattr(
+        config.httpdb.builder, "docker_registry_secret", docker_registry_secret
+    )
```

Comment:
Is a monkeypatch really required?
```suggestion
def test_get_image_dest(dest, registry, default_repository, expected_dest):
    docker_registry_secret = "default-docker-registry-secret"
    config.httpdb.builder.docker_registry = default_repository
    config.httpdb.builder.docker_registry_secret  = docker_registry_secret
```